---
title: "DGE"
author: "Kelterborn"
date: "2024-03-20"
output:
  github_document:
    html_preview: false
    toc: true
always_allow_html: true
editor_options: 
  chunk_output_type: console
knit: (function(input_file, encoding) {
    })
    rmarkdown::render(input_file,output_file= 'Readme.md')
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=300)

```

# 0. Load
##  - Load R librarys
```{r R_update, eval=FALSE, include=FALSE, echo=FALSE}
BiocManager::install()
```

```{r librarys, include=FALSE}

library(DESeq2)
library(GEOquery)
library(tidyverse)
library(CorLevelPlot)
library(gridExtra)
library(viridis)

library(clusterProfiler)
library(ggupset)
library(enrichplot)
library(stats)
library(cowplot)
library(VennDiagram)
library(ggplot2)
library(stringr)
library(rlang)
library(reshape2)
library(readxl)
library(SummarizedExperiment)
library(biomaRt)
library(org.Hs.eg.db)
library(Rfast)
library(xlsx)
library(EnhancedVolcano)
library(PCAtools)
library(rgl)
library(impute)
library(magrittr)     
library(flashClust)

library(gridExtra)
library(patchwork)
library(grid)
library(reshape2)

library(pheatmap)
library(plyr)

library(kableExtra)
library(knitr)

ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
       s <- "S:")
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/RNA-Seq_Kelly_all",sep="/")
gitdir <- paste(dir,"git_RNAseq_Kelly_Hx",sep="/")
data <- paste(dir,"data",sep="/")
dgedir <- paste(gitdir,"2B_DGE",sep="/")

par(mfrow = c(1,1))

```


## - dds
```{r dds}
# load(file=paste(data,"deseq2.dds", sep="/"))

load(file=paste(data,"deseq2_treatment.dds", sep="/"))
dds_t <- dds
load(file=paste(data,"deseq2_condition.dds", sep="/"))
dds_c <- dds

load(file=paste(data,"deseq2_wgcna.dds", sep="/"))

```

# 1. Make results
```{r results}

design(dds)
# ~genotype + treatment + genotype:treatment
# resultsNames(dds)
l <- length(resultsNames(dds))
cbind(resultsNames(dds),c(rep(0,l)))
#      [,1]                        [,2]
# [1,] "Intercept"                 "0" 
# [2,] "genotype_HIF1A_vs_Kelly"   "0" 
# [3,] "genotype_HIF2A_vs_Kelly"   "0" 
# [4,] "genotype_HIF1B_vs_Kelly"   "0" 
# [5,] "treatment_Hx_vs_Nx"        "0" 
# [6,] "genotypeHIF1A.treatmentHx" "0" 
# [7,] "genotypeHIF2A.treatmentHx" "0" 
# [8,] "genotypeHIF1B.treatmentHx" "0" 

condition <- colData(dds)$condition
treatment <- colData(dds)$treatment
genotype <- colData(dds)$genotype

results_list <- {}

#------ Hx vs. Nx (fixed genotype)
# 0,0,0,0,1,0,0,0
## Hx vs. Nx in HIF1A
r1 <- results(dds, contrast = c(0,0,0,0,1,1,0,0))
r1$symbol <- mcols(dds)$symbol
res.Hif1a.Hx.vs.Nx <- r1 #ok
n<- "Hif1a.Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 
# ok
# same as res.Hif1a.Hx.vs.Nx_test <- r2 <- results(dds, contrast = list( c("treatment_Hx_vs_Nx","genotypeHIF1A.treatmentHx")))

## Hx vs. Nx in HIF2A
r1 <- results(dds, contrast = c(0,0,0,0,1,0,1,0))
r1$symbol <- mcols(dds)$symbol
res.Hif2a.Hx.vs.Nx <- r1 #ok
n<- "Hif2a.Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## Hx vs. Nx in HIF1B
r1 <- results(dds, contrast = c(0,0,0,0,1,0,0,1))
r1$symbol <- mcols(dds)$symbol
res.Hif1b.Hx.vs.Nx <- r1 #ok
n<- "Hif1b.Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## Hx vs. Nx in Kelly
r1 <- results(dds, contrast = c(0,0,0,0,1,0,0,0))
r1$symbol <- mcols(dds)$symbol
res.Kelly.Hx.vs.Nx <- r1 #ok
n<- "Kelly.Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

#------  KO vs. Kelly (Nx)
## HIF1A vs. Kelly (Nx)
r1 <- results(dds, contrast = c(0,1,0,0,0,0,0,0))
r1$symbol <- mcols(dds)$symbol
res.Nx.Hif1a.vs.Kelly <- r1
n<- "Nx.Hif1a.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## HIF2A vs. Kelly (Nx)
r1 <- results(dds, contrast = c(0,0,1,0,0,0,0,0))
r1$symbol <- mcols(dds)$symbol
res.Nx.Hif2a.vs.Kelly <- r1
n<- "Nx.Hif2a.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## HIF1B vs. Kelly (Nx)
r1 <- results(dds, contrast = c(0,0,0,1,0,0,0,0))
r1$symbol <- mcols(dds)$symbol
res.Nx.Hif1b.vs.Kelly <- r1 #ok
n<- "Nx.Hif1b.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 


#------ KO vs. Kelly (Hx)
## HIF1A vs. Kelly (Hx)
r1 <- results(dds, contrast = c(0,1,0,0,0,1,0,0))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1a.vs.Kelly <- r1
n<- "Hx.Hif1a.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## HIF2A vs. Kelly (Hx)
r1 <- results(dds, contrast = c(0,0,1,0,0,0,1,0))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif2a.vs.Kelly <- r1
n<- "Hx.Hif2a.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## HIF1B vs. Kelly (Hx)
r1 <- results(dds, contrast = c(0,0,0,1,0,0,0,1))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1b.vs.Kelly <- r1
n<- "Hx.Hif1b.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 


#------ KO vs. KO (Hx) 
# Hif2A vs. Hif1A (Hx)
r1 <- results(dds, contrast = c(0,-1,1,0,0,-1,1,0))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif2a.vs.Hif1a <- r1
n<- "Hx.Hif2a.vs.Hif1a"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

# Hif1B vs. Hif1A (Hx)
r1 <- results(dds, contrast = c(0,-1,0,1,0,-1,0,1))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1b.vs.Hif1a <- r1
n<- "Hx.Hif1b.vs.Hif1a"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

# Hif1B vs. Hif2A (Hx)
r1 <- results(dds, contrast = c(0,0,-1,1,0,0,-1,1))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1b.vs.Hif2a <- r1
n<- "Hx.Hif1b.vs.Hif2a"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

# Hif1B vs. Hifa (Hif1A & Hif2A) (Hx)
r1 <- results(dds, contrast = c(0,-0.5,-0.5,1,0,-0.5,-0.5,1))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1b.vs.Hif12a <- r1
n<- "Hx.Hif1b.vs.Hif12a"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

#Kelly (Hx) vs. Hifs (Hif1A, Hif2A, Hif1b)
r1 <- results(dds, contrast = c(0,1/3,1/3,1/3,0,1/3,1/3,1/3))
r1$symbol <- mcols(dds)$symbol
res.Hx.Kelly.vs.Hif12ab <- r1
n<- "Hx.Kelly.vs.Hif12ab"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

```

### Advanced results troubleshooting
```{r results_trouble, eval=FALSE}
## Hx vs. Nx (all)
# ?

# (Kelly_Hx - Kelly_Nx) + (Hif1a_Hx - Hif1a_Nx) + (Hif2a_Hx - Hif2a_Nx) + (Hif1b_Hx - Hif1b_Nx)
#           5           +         5+6                      5+7                     5+8           /4
# c(0,0,0,0,4*5/4,1/4)

r1 <- results(dds, contrast = c(0,0,0,0,1,-0.25,-0.25,-0.25))
r1$symbol <- mcols(dds)$symbol
res.Hx.vs.Nx <- r1

summary(r1)
b=0
c=0
d=0
e=0
f=0
g=0
h=0
input <- c(0.01726995,0.04838570,-0.06032016,1.00000000,0.24648441,0.20725899,0.09283565)
input <- c(0,0,0,1,0.25,0.25,0.25)

imput0 <- c(0,0,0,0,0,0,0)

lm_pred <- function(input=c(0,0,0,0,0,0,0)){
  # print(paste("0",b,c,d,e,f,g,h,sep=" "))
  if (
    identical(input, imput0) ) { 0 } else {
  r <- results(dds, contrast = c(0,input[1],input[2],input[3],input[4],input[5],input[6],input[7]))
  # print(input)
  fm <- lm(r_t$log2FoldChange ~ r$log2FoldChange)
  # summary(fm)
  summary(fm)$r.squared
  }
  }

plot(r$log2FoldChange,r_t$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm, col = "red")

lm_pred(2)

optimize(lm_pred,
         interval=0.1,
         lower=-1,
         upper=1,
         maximum = TRUE)

# Method "BFGS" is a quasi-Newton method (also known as a variable metric algorithm), specifically that published simultaneously in 1970 by Broyden, Fletcher, Goldfarb and Shanno. This uses function values and gradients to build up a picture of the surface to be optimized.
# 
# Method "CG" is a conjugate gradients method based on that by Fletcher and Reeves (1964) (but with the option of Polak–Ribiere or Beale–Sorenson updates). Conjugate gradient methods will generally be more fragile than the BFGS method, but as they do not store a matrix they may be successful in much larger optimization problems.
# 
# Method "L-BFGS-B" is that of Byrd et. al. (1995) which allows box constraints, that is each variable can be given a lower and/or upper bound. The initial value must satisfy the constraints. This uses a limited-memory modification of the BFGS quasi-Newton method. If non-trivial bounds are supplied, this method will be selected, with a warning.
# 
# Nocedal and Wright (1999) is a comprehensive reference for the previous three methods.
# 
# Method "SANN" is

optim_res1 <- optim(par=c(0,0,0,1,0.25,0,0),
      # lower=-1,upper=+1,
      # control$ndeps=0.1,
      control = list(fnscale = -1),
      method = "BFGS"
      lm_pred)

summary(fm)$r.squared

lm_pred

r_t <- results(dds_t, contrast = c("treatment","Hx","Nx"))
summary(r_t)

resultsNames(dds_c)
colData(dds_c)$condition
r_c <- results(dds_c,
               name = "condition_Kelly_Hx_vs_Kelly_Nx",
               listValues = c(1,-1))
r_c
summary(r_c)

r_c <- results(dds_c,
               contrast = c(0,1/3,-1/4,1/4,-1/4,1/4,-1/4,1/4))
summary(r_c)



names(summary(fm2))

fm1 <- lm(r_t$log2FoldChange ~ r_c$log2FoldChange)
fm2 <- lm(r1$log2FoldChange ~ r_c$log2FoldChange)
fm3 <- lm(r1$log2FoldChange ~ r_t$log2FoldChange)
plot(fm1)
par(mfrow=c(1,3),mar=c(1,1,1,1))
plot(r_c$log2FoldChange,r_t$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm1, col = "red")
plot(r_c$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm2, col = "red")
plot(r_t$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm3, col = "red")


design(dds)

r2_treatment <- results(dds, contrast = c("treatment","Hx","Nx"))
r2_Kelly_Hx_Nx <- results(dds, contrast = c("condition","Kelly_Hx","Kelly_Nx"))
r1_treatment_Hx_vs_Nx <- results(dds, contrast = c(0,0,0,0,0,1,1,1))
r3 <- results(dds, name = "treatment_Hx_vs_Nx")

r1 <- results(dds, contrast = list(c("genotypeHIF1A.treatmentHx","genotypeHIF2A.treatmentHx","genotypeHIF1B.treatmentHx"))
)

?results

design <- ~ treatment + genotype

# Specify the contrast of interest
contrast <- c("genotypeHif1a", "genotypeHif2a")

# Create the contrast matrix
contrast.matrix <- makeContrasts(contrast, levels = colnames(dds))

# Fit the model with the contrast
dds <- DESeqDataSetFromMatrix(countData = countData, colData = colData, design = design)
dds <- DESeq(dds, test = "Wald", coef = contrast.matrix)

# Extract the results
results <- results(dds)

# View the results
head(results)


boxplot(gene ~ treatment,
  data = data.frame(treatment = colData(dds)$treatment, t(assay(ntd))))
                    

r1 <- r1_treatment_Hx_vs_Nx
a <- 0
b <- 0.33

par(mfrow=c(2,5),mar=c(1,1,1,1))

for (a in 0:1){
  for (b in c(0,0.25,0.5,0.75,1)){
    print(paste("a=",a," b=",b))
    if(a+b>0|a+b<0)
r1 <- results(dds, contrast = c(0,0,0,0,a,b,b,b))
fm1 <- lm(r1$log2FoldChange ~ r2_treatment$log2FoldChange)
# fm2 <- lm(r1$log2FoldChange ~ r2_Kelly_Hx_Nx$log2FoldChange)

plot(r2_treatment$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10), main=paste("r2_A vs. ",a,b,sep=","))
abline(fm1, col = "red")

# plot(r2_Kelly_Hx_Nx$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10), main=paste("r2_B vs. ",a,b,sep=","))
# abline(fm2, col = "red")
  }
}

par(mfrow=c(1,1),mar=c(1,1,1,1))

fm1 <- lm(r2_Kelly_Hx_Nx$log2FoldChange ~ r2_treatment$log2FoldChange)

plot(r2_treatment$log2FoldChange,r2_Kelly_Hx_Nx$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10), main=paste("r2_A vs. ",a,b,sep=","))
abline(fm1, col = "red")

plot(r1$log2FoldChange,r3$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10), main=paste("r2_A vs. ",a,b,sep=","))

# r2 <- results(dds, contrast = list( c("genotype_HIF1B_vs_Kelly","genotypeHIF1B.treatmentHx")))

r1 %>% summary()
r2 %>% summary()

fm <- lm(r1$log2FoldChange ~ r2$log2FoldChange)
plot(r2$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm, col = "red")


res.Hx.vs.Nx  <- results(dds, name="treatment_Hx_vs_Nx")

res.Hif1a.vs.Kelly <- results(dds, name="genotype_HIF1A_vs_Kelly")
res.Hif2a.vs.Kelly <- results(dds, name="genotype_HIF2A_vs_Kelly")
res.Hif1b.vs.Kelly <- results(dds, name="genotype_HIF1B_vs_Kelly")



```




# Vulcanos
```{r}



```

