---
title: "DGE"
author: "Kelterborn"
date: "2024-03-20"
output:
  github_document:
    html_preview: false
    toc: true
always_allow_html: true
editor_options: 
  chunk_output_type: console
knit: (function(input_file, encoding) {
    rmarkdown::render(input_file,output_file= 'Readme.md')
    })
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=300)

```

# 0. Load
##  - Load R librarys
```{r R_update, eval=FALSE, include=FALSE, echo=FALSE}
BiocManager::install()
```

```{r librarys, include=FALSE}

library(DESeq2)
library(GEOquery)
library(tidyverse)
library(CorLevelPlot)
library(gridExtra)
library(viridis)

library(clusterProfiler)
library(ggupset)
library(enrichplot)
library(stats)
library(cowplot)
library(VennDiagram)
library(ggplot2)
library(stringr)
library(rlang)
library(reshape2)
library(readxl)
library(SummarizedExperiment)
library(biomaRt)
library(org.Hs.eg.db)
library(Rfast)
library(xlsx)
library(EnhancedVolcano)
library(PCAtools)
library(rgl)
library(impute)
library(magrittr)     
library(flashClust)
library(AnnotationHub)
library(venn)

library(gridExtra)
library(patchwork)
library(grid)
library(reshape2)

library(pheatmap)
library(plyr)

library(kableExtra)
library(knitr)
library(RColorBrewer)

ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
       s <- "S:")
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/RNA-Seq_Kelly_all",sep="/")
gitdir <- paste(dir,"git_RNAseq_Kelly_Hx",sep="/")
data <- paste(dir,"data",sep="/")
dgedir <- paste(gitdir,"2B_DGE",sep="/")

par(mfrow = c(1,1))

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")

```


## - dds
```{r dds}
# load(file=paste(data,"deseq2.dds", sep="/"))

load(file=paste(data,"deseq2_treatment.dds", sep="/"))
dds_t <- dds
load(file=paste(data,"deseq2_condition.dds", sep="/"))
dds_c <- dds

load(file=paste(data,"deseq2_wgcna.dds", sep="/"))

```


## - functions
```{r functions, include=FALSE}
# topgenes
topgenes_f <- function(res,p=0.05,bM=20,l2FC=1){
a <- subset(res, padj < p & baseMean > bM & abs(log2FoldChange) > l2FC)
a <- a[order(a$baseMean, decreasing = T),]
  a$rank.bm <- seq(1:length(rownames(a)))
a <- a[order(a$padj, decreasing = F),]
  a$rank.padj <- seq(1:length(rownames(a)))
a <- a[order(abs(a$log2FoldChange), decreasing = T),]
  a$rank.l2FC <- seq(1:length(rownames(a)))
a$rank.sum <- a$rank.l2FC+a$rank.bm+a$rank.padj
  a <- a[order(a$rank.sum),]
a
}


# Vulcano
Vulcano_SK <- function(n,
                        ntop=100,
                        topcol="royalblue4",
                        hscol="royalblue1",
                        lcol="grey20",
                        xlim=10,
                        ylim=300)
{
res <- results_list[[n]]
top <- topgenes_f(res=res)
l <- nrow(top)

res_shrink <- lfcShrink(dds, res=res, type="ashr")
res_shrink$symbol <- res$symbol

# define limits for significance
# padj (-log10)
# padj = 0.05 -> -log(base=10, 0.05) = 1.3
#
c_plimit <- 50
L2Flimit <- 2
a_L2Flimit <- -c_plimit/(L2Flimit^2)

res_shrink$ishs <- ifelse((a_L2Flimit*(res_shrink$log2FoldChange^2) + c_plimit) < -log(res_shrink$padj, base=10),TRUE,FALSE)
res_shrink$istop <- ifelse(rownames(res_shrink) %in% rownames(top)[1:ntop],TRUE,FALSE)
res <- res_shrink
res <- res[order(res$ishs, decreasing = F),]
res <- res[order(res$istop, decreasing = F),]

# remove nas
res <- res[!is.na(res$padj),]
res <- res[!is.na(res$log2FoldChange),]

# change shape of outliers
shape <- ifelse(abs(res$log2FoldChange) > xlim, 18,
                ifelse(res$padj < 10^-ylim,18,16))
summary(is.na(shape))
# shape[is.na(shape)] <- 2
names(shape)[shape == 18] <- 'out of scale'
names(shape)[shape == 16] <- 'in range'

# move outliers to coord. max.
res$log2FoldChange[res$log2FoldChange > xlim] <- xlim
res$log2FoldChange[res$log2FoldChange < -xlim] <- -xlim
res$padj[res$padj < 10^-ylim] <- 10^-ylim
summary(res$padj < 10^-ylim)

# change colour if in top genes
## according to ishs
keyvals <- ifelse(
    res$ishs == TRUE, topcol,
      'grey')
keyvals[is.na(keyvals)] <- 'grey'
  names(keyvals)[keyvals == topcol] <- 'highly significant'
  names(keyvals)[keyvals == 'grey'] <- 'other'
dim(res)[1] == length(keyvals)

## according to top
keyvals <- ifelse(
    res$istop == TRUE, topcol,
      'grey')
keyvals[is.na(keyvals)] <- 'grey'
  names(keyvals)[keyvals == topcol] <- 'top 100'
  names(keyvals)[keyvals == 'grey'] <- 'other'
dim(res)[1] == length(keyvals)

## combine top & hs
keyvals <- ifelse(
    res$istop == TRUE, topcol,
    ifelse(
    res$ishs == TRUE, hscol,
    'grey') )
    
keyvals[is.na(keyvals)] <- 'grey'
  names(keyvals)[keyvals == topcol] <- 'top 100'
  names(keyvals)[keyvals == hscol] <- 'highly significant'
  names(keyvals)[keyvals == 'grey'] <- 'other'

rownames(res) <- res$symbol

ev_f <- EnhancedVolcano(res,
    x = 'log2FoldChange',
    y = 'padj',
    lab = res$symbol,
    selectLab  = res[res$istop == TRUE,"symbol"],
    labSize = 2,
    drawConnectors = TRUE,
    boxedLabels = TRUE,
    widthConnectors = 0.5,
    colConnectors = lcol,
    pointSize = c(ifelse(res$istop==TRUE, 2, ifelse(res$ishs==TRUE, 1.5, 1))),
    colAlpha = c(ifelse(res$istop==TRUE, 0.9, ifelse(res$ishs==TRUE, 0.5, 0.2))),
    max.overlaps = 15,
    colCustom = keyvals,
    col=c('grey', topcol, topcol, topcol),
    xlim = c(-xlim, xlim),
    ylim = c(0, ylim),
    ylab = "Padj (-log10)",
    title = n,
    subtitle = paste("DE genes:",l),
    # sub = "SVF",
    pCutoff = 10^(-50),
    FCcutoff = 2,
    shapeCustom =shape,
    # pointSize = c(ifelse(rownames(res_WT_D_vs.WT_BL) %in% rownames(top_WT_BL_vs.pcry_BL), 8, 1)),
    legendLabels=c('Not sig.','|L2F| > 2.5','p-adj < 0.05',
                   'p-adj & L2F'),
    legendPosition = 'bottom',
    legendLabSize = 8,
    legendIconSize = 2.0,
    axisLabSize = 8
   )

ev_f
}



```


# 1. Make results
```{r results, include=FALSE}

design(dds)
# ~genotype + treatment + genotype:treatment
# resultsNames(dds)
l <- length(resultsNames(dds))
cbind(resultsNames(dds),c(rep(0,l)))
#      [,1]                        [,2]
# [1,] "Intercept"                 "0" 
# [2,] "genotype_HIF1A_vs_Kelly"   "0" 
# [3,] "genotype_HIF2A_vs_Kelly"   "0" 
# [4,] "genotype_HIF1B_vs_Kelly"   "0" 
# [5,] "treatment_Hx_vs_Nx"        "0" 
# [6,] "genotypeHIF1A.treatmentHx" "0" 
# [7,] "genotypeHIF2A.treatmentHx" "0" 
# [8,] "genotypeHIF1B.treatmentHx" "0" 

condition <- colData(dds)$condition
treatment <- colData(dds)$treatment
genotype <- colData(dds)$genotype

results_list <- {}

#------ Hx vs. Nx (fixed genotype)
# 0,0,0,0,1,0,0,0
## Hx vs. Nx in HIF1A
r1 <- results(dds, contrast = c(0,0,0,0,1,1,0,0))
r1$symbol <- mcols(dds)$symbol
res.Hif1a.Hx.vs.Nx <- r1 #ok
n<- "Hif1a.Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

# ok
# same as res.Hif1a.Hx.vs.Nx_test <- r2 <- results(dds, contrast = list( c("treatment_Hx_vs_Nx","genotypeHIF1A.treatmentHx")))

## Hx vs. Nx in HIF2A
r1 <- results(dds, contrast = c(0,0,0,0,1,0,1,0))
r1$symbol <- mcols(dds)$symbol
res.Hif2a.Hx.vs.Nx <- r1 #ok
n<- "Hif2a.Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## Hx vs. Nx in HIF1B
r1 <- results(dds, contrast = c(0,0,0,0,1,0,0,1))
r1$symbol <- mcols(dds)$symbol
res.Hif1b.Hx.vs.Nx <- r1 #ok
n<- "Hif1b.Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## Hx vs. Nx in Kelly
r1 <- results(dds, contrast = c(0,0,0,0,1,0,0,0))
r1$symbol <- mcols(dds)$symbol
res.Kelly.Hx.vs.Nx <- r1 #ok
n<- "Kelly.Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 


#------  KO vs. Kelly (Nx)
## HIF1A vs. Kelly (Nx)
r1 <- results(dds, contrast = c(0,1,0,0,0,0,0,0))
r1$symbol <- mcols(dds)$symbol
res.Nx.Hif1a.vs.Kelly <- r1
n<- "Nx.Hif1a.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## HIF2A vs. Kelly (Nx)
r1 <- results(dds, contrast = c(0,0,1,0,0,0,0,0))
r1$symbol <- mcols(dds)$symbol
res.Nx.Hif2a.vs.Kelly <- r1
n<- "Nx.Hif2a.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## HIF1B vs. Kelly (Nx)
r1 <- results(dds, contrast = c(0,0,0,1,0,0,0,0))
r1$symbol <- mcols(dds)$symbol
res.Nx.Hif1b.vs.Kelly <- r1 #ok
n<- "Nx.Hif1b.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 


#------ KO vs. Kelly (Hx)
## HIF1A vs. Kelly (Hx)
r1 <- results(dds, contrast = c(0,1,0,0,0,1,0,0))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1a.vs.Kelly <- r1
n<- "Hx.Hif1a.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## HIF2A vs. Kelly (Hx)
r1 <- results(dds, contrast = c(0,0,1,0,0,0,1,0))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif2a.vs.Kelly <- r1
n<- "Hx.Hif2a.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

## HIF1B vs. Kelly (Hx)
r1 <- results(dds, contrast = c(0,0,0,1,0,0,0,1))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1b.vs.Kelly <- r1
n<- "Hx.Hif1b.vs.Kelly"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 


#------ KO vs. KO (Hx) 
# Hif2A vs. Hif1A (Hx)
r1 <- results(dds, contrast = c(0,-1,1,0,0,-1,1,0))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif2a.vs.Hif1a <- r1
n<- "Hx.Hif2a.vs.Hif1a"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1 

# Hif1B vs. Hif1A (Hx)
r1 <- results(dds, contrast = c(0,-1,0,1,0,-1,0,1))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1b.vs.Hif1a <- r1
n<- "Hx.Hif1b.vs.Hif1a"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

# Hif1B vs. Hif2A (Hx)
r1 <- results(dds, contrast = c(0,0,-1,1,0,0,-1,1))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1b.vs.Hif2a <- r1
n<- "Hx.Hif1b.vs.Hif2a"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1


#----ðð KO (Hx.vs.Nx) vs. Kelly (Hx vs. Nx)
# Hif1a vs. Kelly (Hx vs. Nx)
r1 <- results(dds, contrast =  c(0,0,0,0,0,1,0,0))
r1$symbol <- mcols(dds)$symbol
n<- "Hif1aHxNx.vs.KellyHxNx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

# Hif2a vs. Kelly (Hx vs. Nx)
r1 <- results(dds, contrast =  c(0,0,0,0,0,0,1,0))
r1$symbol <- mcols(dds)$symbol
n<- "Hif2aHxNx.vs.KellyHxNx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

# Hif1b vs. Kelly (Hx vs. Nx)
r1 <- results(dds, contrast =  c(0,0,0,0,0,0,0,1))
r1$symbol <- mcols(dds)$symbol
n<- "Hif1bHxNx.vs.KellyHxNx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1


#------ Group comparisions (Hx) 
# Hif1B vs. Hifa (Hif1A & Hif2A) (Hx)
r1 <- results(dds, contrast = c(0,-0.5,-0.5,1,0,-0.5,-0.5,1))
r1$symbol <- mcols(dds)$symbol
res.Hx.Hif1b.vs.Hif12a <- r1
n<- "Hx.Hif1b.vs.Hif12a"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1

# Kelly vs. Hifs (Hif1A, Hif2A, Hif1b) (Hx)
r1 <- results(dds, contrast = c(0,1/3,1/3,1/3,0,1/3,1/3,1/3))
r1$symbol <- mcols(dds)$symbol
res.Hx.Kelly.vs.allHIFs <- r1
n<- "Hx.Kelly.vs.allHIFs"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1


#------ Hx vs. Nx
# Hx vs. Nx over all groups
r1 <- results(dds, contrast = c(0,0,0,0,1,1/4,1/4,1/4))
r1$symbol <- mcols(dds)$symbol
res.Hx.vs.Nx <- r1
n<- "Hx.vs.Nx"
assign(paste("res",n,sep="."),r1)
results_list[[n]] <- r1
# NOT confirmed


results_list[["Nx.Hif2a.vs.Kelly"]] %>% summary(alpha=0.05)
results_list[["Hx.Hif2a.vs.Kelly"]] %>% summary()


```

#### (Advanced results troubleshooting)
```{r results_trouble, eval=FALSE, include=FALSE}
## Hx vs. Nx (all)
# ?

# (Kelly_Hx - Kelly_Nx) + (Hif1a_Hx - Hif1a_Nx) + (Hif2a_Hx - Hif2a_Nx) + (Hif1b_Hx - Hif1b_Nx)
#           5           +         5+6                      5+7                     5+8           /4
# c(0,0,0,0,4*5/4,1/4)

r1 <- results(dds, contrast = c(0,0,0,0,1,-0.25,-0.25,-0.25))
r1$symbol <- mcols(dds)$symbol
res.Hx.vs.Nx <- r1

summary(r1)
b=0
c=0
d=0
e=0
f=0
g=0
h=0
input <- c(0.01726995,0.04838570,-0.06032016,1.00000000,0.24648441,0.20725899,0.09283565)
input <- c(0,0,0,1,0.25,0.25,0.25)

imput0 <- c(0,0,0,0,0,0,0)

lm_pred <- function(input=c(0,0,0,0,0,0,0)){
  # print(paste("0",b,c,d,e,f,g,h,sep=" "))
  if (
    identical(input, imput0) ) { 0 } else {
  r <- results(dds, contrast = c(0,input[1],input[2],input[3],input[4],input[5],input[6],input[7]))
  # print(input)
  fm <- lm(r_t$log2FoldChange ~ r$log2FoldChange)
  # summary(fm)
  summary(fm)$r.squared
  }
  }

plot(r$log2FoldChange,r_t$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm, col = "red")

lm_pred(2)

optimize(lm_pred,
         interval=0.1,
         lower=-1,
         upper=1,
         maximum = TRUE)

# Method "BFGS" is a quasi-Newton method (also known as a variable metric algorithm), specifically that published simultaneously in 1970 by Broyden, Fletcher, Goldfarb and Shanno. This uses function values and gradients to build up a picture of the surface to be optimized.
# 
# Method "CG" is a conjugate gradients method based on that by Fletcher and Reeves (1964) (but with the option of Polak–Ribiere or Beale–Sorenson updates). Conjugate gradient methods will generally be more fragile than the BFGS method, but as they do not store a matrix they may be successful in much larger optimization problems.
# 
# Method "L-BFGS-B" is that of Byrd et. al. (1995) which allows box constraints, that is each variable can be given a lower and/or upper bound. The initial value must satisfy the constraints. This uses a limited-memory modification of the BFGS quasi-Newton method. If non-trivial bounds are supplied, this method will be selected, with a warning.
# 
# Nocedal and Wright (1999) is a comprehensive reference for the previous three methods.
# 
# Method "SANN" is

optim_res1 <- optim(par=c(0,0,0,0.5,0,0,0),
      # lower=-1,upper=+1,
      # control$ndeps=0.1,
      control = list(fnscale = -1),
      # method = "BFGS",
      lm_pred)

optim_res1 <- optim(par=c(0,0,0,0.5,0,0,0),
      control = list(fnscale = -1),
      # method = "BFGS",
      lm_pred)

summary(fm)$r.squared

lm_pred

r_t <- results(dds_t, contrast = c("treatment","Hx","Nx"))
summary(r_t)

resultsNames(dds_c)
colData(dds_c)$condition
r_c <- results(dds_c,
               name = "condition_Kelly_Hx_vs_Kelly_Nx",
               listValues = c(1,-1))
r_c
summary(r_c)

r_c <- results(dds_c,
               contrast = c(0,1/3,-1/4,1/4,-1/4,1/4,-1/4,1/4))
summary(r_c)



names(summary(fm2))

fm1 <- lm(r_t$log2FoldChange ~ r_c$log2FoldChange)
fm2 <- lm(r1$log2FoldChange ~ r_c$log2FoldChange)
fm3 <- lm(r1$log2FoldChange ~ r_t$log2FoldChange)
plot(fm1)
par(mfrow=c(1,3),mar=c(1,1,1,1))
plot(r_c$log2FoldChange,r_t$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm1, col = "red")
plot(r_c$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm2, col = "red")
plot(r_t$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm3, col = "red")


design(dds)

r2_treatment <- results(dds, contrast = c("treatment","Hx","Nx"))
r2_Kelly_Hx_Nx <- results(dds, contrast = c("condition","Kelly_Hx","Kelly_Nx"))
r1_treatment_Hx_vs_Nx <- results(dds, contrast = c(0,0,0,0,0,1,1,1))
r3 <- results(dds, name = "treatment_Hx_vs_Nx")

r1 <- results(dds, contrast = list(c("genotypeHIF1A.treatmentHx","genotypeHIF2A.treatmentHx","genotypeHIF1B.treatmentHx"))
)

?results

design <- ~ treatment + genotype

# Specify the contrast of interest
contrast <- c("genotypeHif1a", "genotypeHif2a")

# Create the contrast matrix
contrast.matrix <- makeContrasts(contrast, levels = colnames(dds))

# Fit the model with the contrast
dds <- DESeqDataSetFromMatrix(countData = countData, colData = colData, design = design)
dds <- DESeq(dds, test = "Wald", coef = contrast.matrix)

# Extract the results
results <- results(dds)

# View the results
head(results)


boxplot(gene ~ treatment,
  data = data.frame(treatment = colData(dds)$treatment, t(assay(ntd))))
                    

r1 <- r1_treatment_Hx_vs_Nx
a <- 0
b <- 0.33

par(mfrow=c(2,5),mar=c(1,1,1,1))

for (a in 0:1){
  for (b in c(0,0.25,0.5,0.75,1)){
    print(paste("a=",a," b=",b))
    if(a+b>0|a+b<0)
r1 <- results(dds, contrast = c(0,0,0,0,a,b,b,b))
fm1 <- lm(r1$log2FoldChange ~ r2_treatment$log2FoldChange)
# fm2 <- lm(r1$log2FoldChange ~ r2_Kelly_Hx_Nx$log2FoldChange)

plot(r2_treatment$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10), main=paste("r2_A vs. ",a,b,sep=","))
abline(fm1, col = "red")

# plot(r2_Kelly_Hx_Nx$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10), main=paste("r2_B vs. ",a,b,sep=","))
# abline(fm2, col = "red")
  }
}

par(mfrow=c(1,1),mar=c(1,1,1,1))

fm1 <- lm(r2_Kelly_Hx_Nx$log2FoldChange ~ r2_treatment$log2FoldChange)

plot(r2_treatment$log2FoldChange,r2_Kelly_Hx_Nx$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10), main=paste("r2_A vs. ",a,b,sep=","))
abline(fm1, col = "red")

plot(r1$log2FoldChange,r3$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10), main=paste("r2_A vs. ",a,b,sep=","))

# r2 <- results(dds, contrast = list( c("genotype_HIF1B_vs_Kelly","genotypeHIF1B.treatmentHx")))

r1 %>% summary()
r2 %>% summary()

fm <- lm(r1$log2FoldChange ~ r2$log2FoldChange)
plot(r2$log2FoldChange,r1$log2FoldChange, xlim = c(-10, 10), ylim = c(-10, 10))
abline(fm, col = "red")


res.Hx.vs.Nx  <- results(dds, name="treatment_Hx_vs_Nx")

res.Hif1a.vs.Kelly <- results(dds, name="genotype_HIF1A_vs_Kelly")
res.Hif2a.vs.Kelly <- results(dds, name="genotype_HIF2A_vs_Kelly")
res.Hif1b.vs.Kelly <- results(dds, name="genotype_HIF1B_vs_Kelly")



```

### -Generate toplist
```{r toplist}
topgenes_list <- lapply(results_list,topgenes_f, p=0.05,bM=50,l2FC=2) %>%  lapply(.,rownames) 
topgenes_list <- lapply(results_list,topgenes_f) %>%  lapply(.,rownames) 

```

```{r results_name}

design(dds)
names(results_list)

```


##  -Plot example counts
```{r, results_counts, fig.height=4}

cols = brewer.pal(n=8,name = 'Paired')

# get 3 sample genes
sample_plots <- list()
sample_plots_list <- {}
plot_list <- list()
li <- results_list %>% length()
for (i in 1:li){
  n <- results_list[i] %>% names()
  goi <- c(topgenes_list[[i]][1],sample(topgenes_list[[i]],size=2))
  l <- length(goi)
       for (ig in 1:l){
  s <- mcols(dds)[goi[ig],"symbol"]
  if (s ==""){s <- goi[ig]}
    d <- plotCounts(dds, gene=goi[ig], intgroup=c("condition","experiment","genotype","treatment"), main=s,returnData=TRUE)

  gcounts <- ggplot(d, aes(x = genotype, y = count, fill=treatment, color=treatment)) +
    geom_boxplot(color="black") +
    geom_point(shape=21,color="black",aes(fill=treatment),position=position_dodge(width=0.75), alpha=1) +
    scale_fill_manual(values=cols[c(1,5)]) +
    scale_color_manual(values=cols[c(1,5)]) +
    scale_y_continuous(trans = "log2") +
    labs(title = paste(s,"(",goi[ig],")",sep=" "))
  assign(paste("gcounts_",ig,sep=""),gcounts)
  
  plot_list[[paste(n,s,sep="_")]] <- gcounts
       }
  
  plot <- gcounts_1 + gcounts_2 + gcounts_3 +  
    plot_annotation(title = n) +
    plot_layout(guides = "collect", axis_titles="collect") & theme(legend.position = 'bottom',plot.title = element_text(size=10))
  print(plot)
}

for (n in names(results_list)){
pn <-   str_detect(names(plot_list), pattern=n) %>% names(plot_list)[.]
plot.3 <- patchwork::wrap_plots(plot_list[pn],ncol = 3) + plot_layout(guides = "collect", axis_titles="collect") + plot_annotation(title = str_split(pn[[1]],pattern="_",simplify =TRUE)[1]) & theme(legend.position = 'bottom',plot.title = element_text(size=10))
#        print(plot.3)
        }            


```


# 2. Data Dive
## Colour sheme
```{r colors}

colors_paired <- c("#A6CEE3", "#1F78B4", "#B2DF8A", "#33A02C", "#FB9A99", "#E31A1C", "#FDBF6F", "#FF7F00")
colors_v <- c("#440154FF", "#482878FF", "#3E4A89FF", "#31688EFF", "#26828EFF", "#1F9E89FF", "#35B779FF", "#6DCD59FF", "#B4DE2CFF", "#FDE725FF")
colors <- c("lavenderblush3","lavenderblush4","#90caf9","#1976d2", "#82e0aa", "#239b56", "#f8c471", "#b9770e") 
colors4 <- colors[c(1,3,5,7)]

```


## Overlaps (Venn)
```{r venn}

# ,fig.show="hold", out.width="80%", fig.with="50%",fig.height=4

# Hx vs. Nx
# dev.off()
input_list <- topgenes_list[c(4,3,1,2)]
plt <- venn.diagram(
    x = input_list,
    category.names = paste(names(topgenes_list[c(4,3,1,2)]),"\n(",topgenes_list %>% summary() %>% .[c(4,3,1,2)],")",sep=""),
    force.unique = TRUE, na = "remove",
    filename = NULL,
    main = "Hx vs. Nx", main.fontface = "bold",
    lwd = 2,
    lty = 'blank',
    fill = colors4[c(1,4,2,3)])

grid.newpage
grid.draw(plt)

# Hx vs. Hx
# dev.off()
plt <- venn.diagram(
    x = topgenes_list[c(8,9,10)],
    category.names = paste(names(topgenes_list[c(8,9,10)]),"\n(",topgenes_list %>% summary() %>% .[c(8,9,10)],")",sep=""),
    force.unique = TRUE, na = "remove",
    filename = NULL,
    main = "Hx vs. Hx", main.fontface = "bold",
    lwd = 2,
    lty = 'blank',
    fill = colors4[c(2,3,4)])
grid.draw(plt)

# Hif1a
# dev.off()
grid.newpage()
plt <- venn.diagram(
    x = topgenes_list[c(4,1,5,8)],
    category.names = paste(names(topgenes_list[c(4,1,5,8)]),"\n(",topgenes_list %>% summary() %>% .[c(4,1,5,8)],")",sep=""),
    force.unique = TRUE, na = "remove",
    filename = NULL,
    main = "Hif1a", main.fontface = "bold",
    lwd = 2,
    lty = 'blank',
    fill = colors[c(2,4,1,3)])
grid.draw(plt)

# Hif1a_simple
# dev.off()
input_list <- topgenes_list[c(4,1,8)]
plt <- venn.diagram(
    x = input_list,
    category.names = paste(names(input_list),"\n(",topgenes_list %>% summary() %>% .[choose_list],")",sep=""),
    force.unique = TRUE, na = "remove",
    filename = NULL,
    main = "Hif1a_simple", main.fontface = "bold",
    lwd = 2,
    lty = 'blank',
    fill = colors[c(1,3,4)])

overlaps <- calculate.overlap(input_list)
lapply(overlaps,length)

l <- plt %>% length()
for (i in 1:l){
  print(i)
print (plt[[i]]$label == lapply(overlaps,length))
}

## label names
plt[[7]]$label <- paste(plt[[7]]$label,"a1",sep="\n")
plt[[9]]$label <- paste(plt[[9]]$label,"a2",sep="\n")

plt[[9]]$label <- paste(overlaps[[1]], collapse = "\n") 

plt[[10]]$label <- paste(overlaps[[6]], collapse = "\n") 

grid.newpage()
grid.draw(plt)

# Hif1a_all
# dev.off()
choose_list <- c(4,14,1,8)
plt <- venn.diagram(
    x = topgenes_list[choose_list],
    category.names = paste(names(topgenes_list[choose_list]),"\n(",topgenes_list %>% summary() %>% .[choose_list],")",sep=""),
    force.unique = TRUE, na = "remove",
    filename = NULL,
    main = "Hif1a_all", main.fontface = "bold",
    lwd = 2,
    lty = 'blank',
    fill = colors[c(2,4,1,3)])
grid.newpage()
grid.draw(plt)

# Hif2a
# dev.off()
plt <- venn.diagram(
    x = topgenes_list[c(4,2,6,9)],
    category.names = paste(names(topgenes_list[c(4,2,6,9)]),"\n(",topgenes_list %>% summary() %>% .[c(4,2,6,9)],")",sep=""),
    force.unique = TRUE, na = "remove",
    filename = NULL,
    main = "Hif2a", main.fontface = "bold",
    lwd = 2,
    lty = 'blank',
    fill = colors[c(2,6,1,5)])
grid.newpage()
grid.draw(plt)

# Hif1b
# dev.off()
plt <- venn.diagram(
    x = topgenes_list[c(4,3,7,10)],
    category.names = paste(names(topgenes_list[c(4,3,7,10)]),"\n(",topgenes_list %>% summary() %>% .[c(4,3,7,10)],")",sep=""),
    force.unique = TRUE, na = "remove",
    filename = NULL,
    main = "Hif2a", main.fontface = "bold",
    lwd = 2,
    lty = 'blank',
    fill = colors[c(2,8,1,7)])
grid.newpage()
grid.draw(plt)

```




```{r}



```


## Volcanos
### prepare data
```{r volcano_data, eval=FALSE, include=FALSE}
names(results_list)

# colours
topcol <- "royalblue4"
hscol <- "royalblue1"
lcol <- "grey20"

# limits
xlim <- 10
ylim <- 300

# number of top genes
ntop <- 100

# Kelly (Hx vs. Nx)
n <- "Kelly.Hx.vs.Nx"



res <- results_list[[n]]
top <- topgenes_f(res)
l <- nrow(top)

res_shrink <- lfcShrink(dds, res=res, type="ashr")
res_shrink$symbol <- res$symbol 
res_shrink$ishs <- ifelse((-20/3*(res_shrink$log2FoldChange^2) + 60) < -log(res_shrink$padj, base=10),TRUE,FALSE)
res_shrink$istop <- ifelse(rownames(res_shrink) %in% rownames(top)[1:ntop],TRUE,FALSE)
res <- res_shrink
res <- res[order(res$ishs, decreasing = F),]
res <- res[order(res$istop, decreasing = F),]

# remove nas
res <- res[!is.na(res$padj),]
res <- res[!is.na(res$log2FoldChange),]

# change shape of outliers
shape <- ifelse(abs(res$log2FoldChange) > xlim, 18,
                ifelse(res$padj < 10^-ylim,18,16))
summary(is.na(shape))
# shape[is.na(shape)] <- 2
names(shape)[shape == 18] <- 'out of scale'
names(shape)[shape == 16] <- 'in range'

# move outliers to coord. max.
res$log2FoldChange[res$log2FoldChange > xlim] <- xlim
res$log2FoldChange[res$log2FoldChange < -xlim] <- -xlim
res$padj[res$padj < 10^-ylim] <- 10^-ylim
summary(res$padj < 10^-ylim)

# change colour if in top genes
## according to ishs
keyvals <- ifelse(
    res$ishs == TRUE, topcol,
      'grey')
keyvals[is.na(keyvals)] <- 'grey'
  names(keyvals)[keyvals == topcol] <- 'highly significant'
  names(keyvals)[keyvals == 'grey'] <- 'other'
dim(res)[1] == length(keyvals)

## according to top
keyvals <- ifelse(
    res$istop == TRUE, topcol,
      'grey')
keyvals[is.na(keyvals)] <- 'grey'
  names(keyvals)[keyvals == topcol] <- 'top 100'
  names(keyvals)[keyvals == 'grey'] <- 'other'
dim(res)[1] == length(keyvals)

## combine top & hs
keyvals <- ifelse(
    res$istop == TRUE, topcol,
    ifelse(
    res$ishs == TRUE, hscol,
    'grey') )
    
keyvals[is.na(keyvals)] <- 'grey'
  names(keyvals)[keyvals == topcol] <- 'top 100'
  names(keyvals)[keyvals == hscol] <- 'highly significant'
  names(keyvals)[keyvals == 'grey'] <- 'other'
dim(res)[1] == length(keyvals)


rownames(res) <- res$symbol

```

### simple volcano (full)
```{r volcano full, eval=FALSE, include=FALSE}
length(rownames(res))==length(shape)
length(rownames(res))==length(keyvals)

ev <- EnhancedVolcano(res,
    x = 'log2FoldChange',
    y = 'padj',
    lab = res$symbol,
    selectLab  = res[res$istop == TRUE,"symbol"],
    labSize = 3,
    drawConnectors = TRUE,
    boxedLabels = TRUE,
    widthConnectors = 0.5,
    colConnectors = lcol,
    pointSize = c(ifelse(res$istop==TRUE, 4, ifelse(res$ishs==TRUE, 3, 2))),
    colAlpha = c(ifelse(res$istop==TRUE, 0.9, ifelse(res$ishs==TRUE, 0.5, 0.2))),
    max.overlaps = 10,
    colCustom = keyvals,
    col=c('grey', topcol, topcol, topcol),
    xlim = c(-xlim, xlim),
    ylim = c(0, ylim),
    ylab = "Padj (-log10)",
    title = n,
    subtitle = paste("DE genes:",l),
    # sub = "SVF",
    pCutoff = 10^(-60),
    FCcutoff = 2,
   shapeCustom =shape,
    # pointSize = c(ifelse(rownames(res_WT_D_vs.WT_BL) %in% rownames(top_WT_BL_vs.pcry_BL), 8, 1)),
    legendLabels=c('Not sig.','|L2F| > 2.5','p-adj < 0.05',
                   'p-adj & L2F'),
    legendPosition = 'bottom'
    )
ev

```

#### check cutoff
```{r check_cutoff, eval=FALSE, include=FALSE}
a=a_L2Flimit
b=0
c=plimit

f = function(x){
  a*x^2 + b*x + c
}

x = -10:10
plot(x,f(x),type='l',ylim=c(0,100))
abline(h=0)
abline(v=0)

find.roots = function(a, b, c) {
  discriminant = b^2 - 4 * a * c
  if (discriminant > 0) {
    c((-b - sqrt(discriminant))/(2 * a), (-b + sqrt(discriminant))/(2 * a))
  }
  else if (discriminant == 0) {
    -b / (2 * a)
  }
  else {
    NaN
  }
}
solutions = find.roots(a, b, c)

points(x = solutions, y = rep(0, length(solutions)), # x and y coordinates of the x-intercepts
       pch = 18, cex = 2, col = 'red')
text(x = solutions, y = rep(0, length(solutions)),
     labels = rep("x-intercept", length(solutions)),
     pos = 3, col = 'red')



points(x=res$log2FoldChange,
       y=-log(base=10,res$padj),
       pch=16,col='royalblue',cex=0.5)

plot(x,-log(base=10,x))

```


### Draw Vulcanos
```{r draw vulcano,  fig.show="hold", fig.height=4}

# Input

# results name
n <- "Hif1a.Hx.vs.Nx"

# colours
topcol <- "royalblue4"
hscol <- "royalblue1"
lcol <- "grey20"

# limits
xlim <- 10
ylim <- 300

# number of top genes
ntop <- 100

###################

# names(results_list)

ev_kelly <- Vulcano_SK(n="Kelly.Hx.vs.Nx", # results name
            ntop=200, # number of top genes
            topcol=colors[4], # color top genes
            hscol=colors[3], # color highly significant genes
            lcol="grey20",
            xlim=10,
            ylim=300)

ev_hif1a <- Vulcano_SK(n <- "Hif1a.Hx.vs.Nx",
            ntop=200,
            topcol="orangered3",
            hscol="salmon1",
            lcol="grey20")

ev_hif2a <- Vulcano_SK(n <- "Hif2a.Hx.vs.Nx",
            ntop=200,
            topcol="hotpink4",
            hscol="hotpink1",
            lcol="grey20")

ev_hif1b <- Vulcano_SK(n <- "Hif1b.Hx.vs.Nx",
            ntop=200,
            topcol="darkseagreen4",
            hscol="darkseagreen1",
            lcol="grey20")

( ev_kelly + ev_hif1b ) + plot_layout(guides = "collect", axes="collect", axis_titles="collect") & theme(legend.position = 'bottom', axis.title=element_text(size=8))
( ev_hif1a + ev_hif2a) + plot_layout(guides = "collect", axis_titles="collect") & theme(legend.position = 'bottom')
 
```

