---
title: "DGE"
author: "Kelterborn"
date: "2024-03-20"
output:
  md_document:
    variant: gfm
    toc: true
always_allow_html: true
editor_options: 
  chunk_output_type: console
knit: (function(inputFile, encoding) {
    rmarkdown::render(inputFile, 
      encoding = encoding, 
      output_file = paste0(dirname(inputFile),'/README.md')) })
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=300)

```

# 0. Load
##  - R_libraries 
BiocManager::install()

```{r librarys, include=FALSE}

library(DESeq2)
library(GEOquery)
library(tidyverse)
library(CorLevelPlot)
library(gridExtra)
library(viridis)

library(clusterProfiler)
library(ggupset)
library(enrichplot)
library(stats)
library(cowplot)
library(VennDiagram)
library(ggplot2)
library(stringr)
library(rlang)
library(reshape2)
library(readxl)
library(SummarizedExperiment)
library(biomaRt)
library(org.Hs.eg.db)
library(Rfast)
library(xlsx)
library(EnhancedVolcano)
library(PCAtools)
library(rgl)
library(impute)
library(magrittr)     
library(flashClust)

library(gridExtra)
library(patchwork)
library(grid)
library(reshape2)

library(pheatmap)
library(plyr)

library(kableExtra)
library(knitr)

ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
       s <- "S:")
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/RNA-Seq_Kelly_all",sep="/")
gitdir <- paste(dir,"git_RNAseq_Kelly_Hx",sep="/")
data <- paste(dir,"data",sep="/")
dgedir <- paste(gitdir,"2B_DGE",sep="/")


```


## - dds
```{r dds}
load(file=paste(data,"deseq2.dds", sep="/"))

```

# 1. Make results
```{r results}
design(dds)
# ~genotype + treatment + genotype:treatment
resultsNames(dds)
# [1] "Intercept"                 "genotype_HIF1A_vs_Kelly"   "genotype_HIF2A_vs_Kelly"   "genotype_HIF1B_vs_Kelly"   "treatment_Hx_vs_Nx"        "genotypeHIF1A.treatmentHx"
# [7] "genotypeHIF2A.treatmentHx" "genotypeHIF1B.treatmentHx"
l <- length(resultsNames(dds))
cbind(resultsNames(dds),c(rep(0,l)))
#      [,1]                        [,2]
# [1,] "Intercept"                 "0" 
# [2,] "genotype_HIF1A_vs_Kelly"   "0" 
# [3,] "genotype_HIF2A_vs_Kelly"   "0" 
# [4,] "genotype_HIF1B_vs_Kelly"   "0" 
# [5,] "treatment_Hx_vs_Nx"        "0" 
# [6,] "genotypeHIF1A.treatmentHx" "0" 
# [7,] "genotypeHIF2A.treatmentHx" "0" 
# [8,] "genotypeHIF1B.treatmentHx" "0" 

condition <- colData(dds)$condition

res.Kelly.Hx.vs.Nx <- results(dds, contrast = c(0,0,0,0,5,0,0,0)) %>% cbind(.,mcols(dds)$symbol)
res.Kelly.Hx.vs.Nx %>% summary()



res.Kelly.Hx.vs.Nx_test <- results(dds)
res.Kelly.Hx.vs.Nx_test %>% summary()

fm <- lm(m.res_WT_D_vs.WT_BL$log2FoldChange ~ res_WT_D_vs.WT_BL$log2FoldChange)
fm
plot(m.res_WT_D_vs.WT_BL$log2FoldChange,res_WT_D_vs.WT_BL$log2FoldChange, xlim = c(-5, 5), ylim = c(-5, 5))
abline(fm, col = "red")

res.Hif1a.Hx.vs.Nx <- results(dds, contrast = c("condition",condition[c(3,3)])) %>% cbind(.,mcols(dds)$symbol)
res.Hif2a.Hx.vs.Nx <- results(dds, contrast = c("condition",condition[c(6,5)])) %>% cbind(.,mcols(dds)$symbol)
res.Hif1b.Hx.vs.Nx <- results(dds, contrast = c("condition",condition[c(8,7)])) %>% cbind(.,mcols(dds)$symbol)

res.Hx.Hif1a.vs.Kelly <- results(dds, contrast = c("condition",condition[c(4,2)])) %>% cbind(.,mcols(dds)$symbol)
res.Hx.Hif2a.vs.Kelly <- results(dds, contrast = c("condition",condition[c(6,2)])) %>% cbind(.,mcols(dds)$symbol)
res.Hx.Hif1b.vs.Kelly <- results(dds, contrast = c("condition",condition[c(8,2)])) %>% cbind(.,mcols(dds)$symbol)

res.Nx.Hif1a.vs.Kelly <- results(dds, contrast = c("condition",condition[c(3,1)])) %>% cbind(.,mcols(dds)$symbol)
res.Nx.Hif2a.vs.Kelly <- results(dds, contrast = c("condition",condition[c(5,2)])) %>% cbind(.,mcols(dds)$symbol)
res.Nx.Hif1b.vs.Kelly <- results(dds, contrast = c("condition",condition[c(7,2)])) %>% cbind(.,mcols(dds)$symbol)

res.Hx.vs.Nx  <- results(dds, name="treatment_Hx_vs_Nx")

res.Hif1a.vs.Kelly <- results(dds, name="genotype_HIF1A_vs_Kelly")
res.Hif2a.vs.Kelly <- results(dds, name="genotype_HIF2A_vs_Kelly")
res.Hif1b.vs.Kelly <- results(dds, name="genotype_HIF1B_vs_Kelly")







```

# Vulcanos
```{r}



```

