---
title: "RNA-Seq Kelly Hx 3A ChIP-Seq"
author: "Kelterborn"
date: "2024-06-19"
output:
  github_document:
    html_preview: false
    toc: true
    toc_depth: 2
always_allow_html: true
editor_options: 
  chunk_output_type: console
knit: (function(input_file, encoding) {
    rmarkdown::render(input_file,output_file= 'Readme.md')
    })
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=200)

setwd(dirname(rstudioapi::getActiveDocumentContext()$path))
allnew <- TRUE

```

# 0. Load

## - Load R librarys

```{r R_update, eval=FALSE, include=FALSE}
BiocManager::install()
BiocManager::valid()
# update.packages(oldPkgs = old.packages())
BiocManager::install("tsne")
install.packages("reticulate")

  BiocManager::install(c(
    "cli", "crayon"
  ), update = TRUE, ask = FALSE, force = TRUE)

```

```{r librarys, include=FALSE}

library(readxl)
library(tidyverse)
library(kableExtra)
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene


requireNamespace('png', quietly = TRUE)

ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
       s <- "S:")
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/RNA-Seq_Kelly_all",sep="/")
gitdir <- paste(dir,"git_RNAseq_Kelly_Hx",sep="/")
data <- paste(dir,"data",sep="/")
pdir <- paste(gitdir,basename(dirname(rstudioapi::getActiveDocumentContext()$path)),sep="/") 
chipdir <- paste0(pdir,"/chip_seq_data",sep="/")
par(mfrow = c(1,1))

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")

```

## - Load dds

```{r dds, include=FALSE}

load(file=paste(data,"deseq2_wgcna.dds", sep="/"))

```

## - functions

```{r topgenes, include=FALSE}
# topgenes
topgenes_f <- function(res,p=0.05,bM=10,l2FC=1){
a <- subset(res, padj < p & baseMean > bM & abs(log2FoldChange) > l2FC)
a <- a[order(a$baseMean, decreasing = T),]
  a$rank.bm <- seq(1:length(rownames(a)))
a <- a[order(a$padj, decreasing = F),]
  a$rank.padj <- seq(1:length(rownames(a)))
a <- a[order(abs(a$log2FoldChange), decreasing = T),]
  a$rank.l2FC <- seq(1:length(rownames(a)))
a$rank.sum <- a$rank.l2FC+a$rank.bm+a$rank.padj
  a <- a[order(a$rank.sum),]
a
}

```

# Overview ChIP-Seq datasets

Literature

+---------------------------+---------------------------------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------+
| Author                    | cells                                       | Ab                  | link                                                                                                                      |
+===========================+=============================================+=====================+===========================================================================================================================+
| **SchÃ¶del et al., 2011**  | MCF-7 (breast)                              | Hif1A, Hif2A, Hif1B | <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3374576/>                                                                   |
+---------------------------+---------------------------------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------+
| **Andrysik et al., 2021** | HCT116 (colon), RKO (colon), A549, and H460 | HIF1A               | <https://www.nature.com/articles/s41467-021-21687-2#Sec11> <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE145157> |
+---------------------------+---------------------------------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------+
| **James A Smythies**      | HKC-8, RCC4, HepG2                          | HIF1A, HIF2A, HIF1B | <https://www.embopress.org/doi/pdf/10.15252/embr.201846401> GSE120885, GSE120886 and GSE120887                            |
|                           |                                             |                     |                                                                                                                           |
|                           |                                             |                     | <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120887>                                                            |
+---------------------------+---------------------------------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------+

ChIP Seq in BE(2)
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167477

# ReMAP

**ReMAP:**

Hif1a: <https://remap.univ-amu.fr/target_page/HIF1A:9606>

Hif2a: <https://remap.univ-amu.fr/target_page/EPAS1:9606>

Hif1b: <https://remap.univ-amu.fr/target_page/ARNT:9606>

## Load dataset
```{r load_chips, echo=FALSE, include=TRUE}
list.files(pdir)
hif1a_datasets <- read_xlsx(paste(pdir,"ReMAP_ChIP_Hif1a.xlsx",sep="/"), )
hif2a_datasets <- read_xlsx(paste(pdir,"ReMAP_ChIP_Hif2a.xlsx",sep="/"), )
hif1b_datasets <- read_xlsx(paste(pdir,"ReMAP_ChIP_Hif1b.xlsx",sep="/"), )

celllines <- data.frame("Hif1A" = (hif1a_datasets$Biotype %>% factor() %>% levels() %>% paste(collapse = " ")),
                       "Hif2A" = (hif2a_datasets$Biotype %>% factor() %>% levels() %>% paste(collapse = " ")),
                       "Hif1B" = (hif1b_datasets$Biotype %>% factor() %>% levels() %>% paste(collapse = " "))) %>% t()

celllines %>% kable()

```

## HIF1A
```{r remap_hif1a}
remap_hif1a <- paste0(chipdir,"/hif1a")
n <- "remap_hif1a"
list.files(remap_hif1a)

run <- "ReMap_hif1a"
name <- paste(run,"_",sep="")

beddir <- paste(remap_hif1a)

beds <- list.files(beddir, pattern=".bed.gz")
beds <- file.path(beddir,beds)

peak <- readPeakFile(beds[2])
length(peak)

peak_table <- {}
i <- beds[1]
for (i in beds) {
peak <- readPeakFile(i)
n <- basename(i) %>% str_remove(pattern=".bed.gz")
print(length(peak))
peak_table <- rbind(peak_table,data.frame(sample = n, peaks = length(peak)))
}
peak_table


promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
genebody <- getBioRegion(TxDb = txdb,
                         by = "gene",
                         type = "body",
                         upstream = rel(1),
                         downstream = rel(1))

file.exists(beds) %>% summary()

peak_list <- lapply(beds,readPeakFile)
peak_list <- lapply(peak_list,keepStandardChromosomes,pruning.mode="coarse")

tagMatrixList_proms <- lapply(peak_list, getTagMatrix, windows=promoter)

tagMatrixList_genes <- lapply(beds, getTagMatrix, windows = genebody, nbin = 100,
                              upstream = rel(1),downstream = rel(1))


names <- paste(name,str_split(basename(beds),pattern="_", simplify = T)[,2], sep="")

names(tagMatrixList_proms) <- c(names)
names(tagMatrixList_genes) <- c(names)

plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000))
plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), facet="row")
plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), conf=0.95, resample=500)

#ggsave(file=paste(csrundir,"/",name,"AvgProf_TSS.pdf",sep=""), width = 8, height = 8, device = "pdf")

plotPeakProf(tagMatrixList_genes)
plotPeakProf(tagMatrixList_genes, conf=0.95, resample=500) #, facet="row"

#ggsave(file=paste(csrundir,"/",name,"PeakProf_gene.pdf",sep=""), width = 8, height = 8, device = "pdf")

tagHeatmap(tagMatrixList_proms)

#ggsave(file=paste(csrundir,"/",name,"Heatmap.pdf",sep=""), width = 12, height = 8, device = "pdf")

peakAnnoList <- lapply(beds, annotatePeak, TxDb=txdb,
                          tssRegion=c(-3000, 3000), verbose=TRUE)

names(peakAnnoList) <- c(names)

plotAnnoBar(peakAnnoList)
#ggsave(file=paste(csrundir,"/",name,"AnnoBar.pdf",sep=""), width = 12, height = 8, device = "pdf")

plotDistToTSS(peakAnnoList)
#ggsave(file=paste(csrundir,"/",name,"DistToTSS.pdf",sep=""), width = 12, height = 8, device = "pdf")





```


# SK-N-BE(2)
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167477

```{r sknbe2}
SK_N_BE2 <- paste0(chipdir,"/SKNBE2")
list.files(SK_N_BE2)

run <- "SK_N_BE2"
name <- paste(run,"_",sep="")

beddir <- paste(remap_hif1a)

beds <- list.files(beddir, pattern=".bed.gz")
beds <- file.path(beddir,beds)

peak <- readPeakFile(beds[2])
length(peak)

peak_table <- {}
i <- beds[1]
for (i in beds) {
peak <- readPeakFile(i)
n <- basename(i) %>% str_remove(pattern=".bed.gz")
print(length(peak))
peak_table <- rbind(peak_table,data.frame(sample = n, peaks = length(peak)))
}
peak_table


promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
genebody <- getBioRegion(TxDb = txdb,
                         by = "gene",
                         type = "body",
                         upstream = rel(1),
                         downstream = rel(1))

file.exists(beds) %>% summary()

peak_list <- lapply(beds,readPeakFile)
peak_list <- lapply(peak_list,keepStandardChromosomes,pruning.mode="coarse")

tagMatrixList_proms <- lapply(peak_list, getTagMatrix, windows=promoter)

tagMatrixList_genes <- lapply(beds, getTagMatrix, windows = genebody, nbin = 100,
                              upstream = rel(1),downstream = rel(1))


names <- paste(name,str_split(basename(beds),pattern="_", simplify = T)[,2], sep="")

names(tagMatrixList_proms) <- c(names)
names(tagMatrixList_genes) <- c(names)

plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000))
plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), facet="row")
plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), conf=0.95, resample=500)

#ggsave(file=paste(csrundir,"/",name,"AvgProf_TSS.pdf",sep=""), width = 8, height = 8, device = "pdf")

plotPeakProf(tagMatrixList_genes)
plotPeakProf(tagMatrixList_genes, conf=0.95, resample=500) #, facet="row"

#ggsave(file=paste(csrundir,"/",name,"PeakProf_gene.pdf",sep=""), width = 8, height = 8, device = "pdf")

tagHeatmap(tagMatrixList_proms)

#ggsave(file=paste(csrundir,"/",name,"Heatmap.pdf",sep=""), width = 12, height = 8, device = "pdf")

peakAnnoList <- lapply(beds, annotatePeak, TxDb=txdb,
                          tssRegion=c(-3000, 3000), verbose=TRUE)

names(peakAnnoList) <- c(names)

plotAnnoBar(peakAnnoList)
#ggsave(file=paste(csrundir,"/",name,"AnnoBar.pdf",sep=""), width = 12, height = 8, device = "pdf")

plotDistToTSS(peakAnnoList)
#ggsave(file=paste(csrundir,"/",name,"DistToTSS.pdf",sep=""), width = 12, height = 8, device = "pdf")




```


# SchÃ¶del
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167477

```{r schoedel}
SK_N_BE2 <- paste0(chipdir,"/SchÃ¶del")
list.files(SK_N_BE2)

run <- "SchÃ¶del"
name <- paste(run,"_",sep="")

beddir <- paste(remap_hif1a)

beds <- list.files(beddir, pattern=".bed.gz")
beds <- file.path(beddir,beds)

peak <- readPeakFile(beds[2])
length(peak)

peak_table <- {}
i <- beds[1]
for (i in beds) {
peak <- readPeakFile(i)
n <- basename(i) %>% str_remove(pattern=".bed.gz")
print(length(peak))
peak_table <- rbind(peak_table,data.frame(sample = n, peaks = length(peak)))
}
peak_table


promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
genebody <- getBioRegion(TxDb = txdb,
                         by = "gene",
                         type = "body",
                         upstream = rel(1),
                         downstream = rel(1))

file.exists(beds) %>% summary()

peak_list <- lapply(beds,readPeakFile)
peak_list <- lapply(peak_list,keepStandardChromosomes,pruning.mode="coarse")

tagMatrixList_proms <- lapply(peak_list, getTagMatrix, windows=promoter)

tagMatrixList_genes <- lapply(beds, getTagMatrix, windows = genebody, nbin = 100,
                              upstream = rel(1),downstream = rel(1))


names <- paste(name,str_split(basename(beds),pattern="_", simplify = T)[,2], sep="")

names(tagMatrixList_proms) <- c(names)
names(tagMatrixList_genes) <- c(names)

plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000))
plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), facet="row")
plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), conf=0.95, resample=500)

#ggsave(file=paste(csrundir,"/",name,"AvgProf_TSS.pdf",sep=""), width = 8, height = 8, device = "pdf")

plotPeakProf(tagMatrixList_genes)
plotPeakProf(tagMatrixList_genes, conf=0.95, resample=500) #, facet="row"

#ggsave(file=paste(csrundir,"/",name,"PeakProf_gene.pdf",sep=""), width = 8, height = 8, device = "pdf")

tagHeatmap(tagMatrixList_proms)

#ggsave(file=paste(csrundir,"/",name,"Heatmap.pdf",sep=""), width = 12, height = 8, device = "pdf")

peakAnnoList <- lapply(beds, annotatePeak, TxDb=txdb,
                          tssRegion=c(-3000, 3000), verbose=TRUE)

names(peakAnnoList) <- c(names)

plotAnnoBar(peakAnnoList)
#ggsave(file=paste(csrundir,"/",name,"AnnoBar.pdf",sep=""), width = 12, height = 8, device = "pdf")

plotDistToTSS(peakAnnoList)
#ggsave(file=paste(csrundir,"/",name,"DistToTSS.pdf",sep=""), width = 12, height = 8, device = "pdf")




```

