---
title: "RNA-Seq Kelly Hx 3A ChIP-Seq"
author: "Kelterborn"
date: "2024-06-19"
output:
  github_document:
    html_preview: false
    toc: true
    toc_depth: 2
always_allow_html: true
editor_options: 
  chunk_output_type: inline
knit: (function(input_file, encoding) {
    rmarkdown::render(input_file,output_file= 'Readme.md')
        })
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=200)

setwd(dirname(rstudioapi::getActiveDocumentContext()$path))
allnew <- TRUE

```

# 0. Load

## - Load R librarys

```{r R_update, eval=FALSE, include=FALSE}
BiocManager::install()
BiocManager::valid()
# update.packages(oldPkgs = old.packages())
BiocManager::install("tsne")
install.packages("reticulate")

  BiocManager::install(c(
    "cli", "crayon"
  ), update = TRUE, ask = FALSE, force = TRUE)

```

```{r librarys, include=FALSE}

library(readxl)
library(tidyverse)
library(kableExtra)
library(ChIPseeker)
library(ChIPpeakAnno)
library(rtracklayer)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene

library(AnnotationHub)
library(VennDiagram)
library(venn)
library(viridis)

requireNamespace('png', quietly = TRUE)

ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
       s <- "S:")
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/RNA-Seq_Kelly_all",sep="/")
gitdir <- paste(dir,"git_RNAseq_Kelly_Hx",sep="/")
data <- paste(dir,"data",sep="/")
pdir <- paste(gitdir,basename(dirname(rstudioapi::getActiveDocumentContext()$path)),sep="/") 
chipdir <- paste0(pdir,"/chip_seq_data",sep="/")
par(mfrow = c(1,1))

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")

# load blacklist
blacklhs<-"/mnt/s/AG/AG-Scholz-NGS/Daten/Simon/Genomic_data/Human/hg38-blacklist.v2.bed.gz"
bl <- readPeakFile(blacklhs)

# Load AnnotationHub
ah = AnnotationHub()
query(ah, c("EnsDb", "sapiens"))
edb <- ah[["AH116860"]]

```

## - Load dds

```{r dds, include=FALSE}

load(file=paste(data,"deseq2_wgcna.dds", sep="/"))

```

## - functions

```{r topgenes, include=FALSE}
# topgenes
topgenes_f <- function(res,p=0.05,bM=10,l2FC=1){
a <- subset(res, padj < p & baseMean > bM & abs(log2FoldChange) > l2FC)
a <- a[order(a$baseMean, decreasing = T),]
  a$rank.bm <- seq(1:length(rownames(a)))
a <- a[order(a$padj, decreasing = F),]
  a$rank.padj <- seq(1:length(rownames(a)))
a <- a[order(abs(a$log2FoldChange), decreasing = T),]
  a$rank.l2FC <- seq(1:length(rownames(a)))
a$rank.sum <- a$rank.l2FC+a$rank.bm+a$rank.padj
  a <- a[order(a$rank.sum),]
a
}

```

# Overview ChIP-Seq datasets

Literature

+---------------------------+---------------------------------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------+
| Author                    | cells                                       | Ab                  | link                                                                                                                      |
+===========================+=============================================+=====================+===========================================================================================================================+
| **Schödel et al., 2011**  | MCF-7 (breast)                              | Hif1A, Hif2A, Hif1B | <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3374576/>                                                                   |
+---------------------------+---------------------------------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------+
| **Andrysik et al., 2021** | HCT116 (colon), RKO (colon), A549, and H460 | HIF1A               | <https://www.nature.com/articles/s41467-021-21687-2#Sec11> <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE145157> |
+---------------------------+---------------------------------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------+
| **James A Smythies**      | HKC-8, RCC4, HepG2                          | HIF1A, HIF2A, HIF1B | <https://www.embopress.org/doi/pdf/10.15252/embr.201846401> GSE120885, GSE120886 and GSE120887                            |
|                           |                                             |                     |                                                                                                                           |
|                           |                                             |                     | <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120887>                                                            |
+---------------------------+---------------------------------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------+

ChIP Seq in BE(2)
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167477

# ReMAP

**ReMAP:**

Hif1a: <https://remap.univ-amu.fr/target_page/HIF1A:9606>

Hif2a: <https://remap.univ-amu.fr/target_page/EPAS1:9606>

Hif1b: <https://remap.univ-amu.fr/target_page/ARNT:9606>

## Load dataset
```{r load_chips, echo=FALSE, include=TRUE}
list.files(pdir)
hif1a_datasets <- read_xlsx(paste(pdir,"ReMAP_ChIP_Hif1a.xlsx",sep="/"), )
hif2a_datasets <- read_xlsx(paste(pdir,"ReMAP_ChIP_Hif2a.xlsx",sep="/"), )
hif1b_datasets <- read_xlsx(paste(pdir,"ReMAP_ChIP_Hif1b.xlsx",sep="/"), )

celllines <- data.frame("Hif1A" = (hif1a_datasets$Biotype %>% factor() %>% levels() %>% paste(collapse = " ")),
                       "Hif2A" = (hif2a_datasets$Biotype %>% factor() %>% levels() %>% paste(collapse = " ")),
                       "Hif1B" = (hif1b_datasets$Biotype %>% factor() %>% levels() %>% paste(collapse = " "))) %>% t()

celllines %>% kable()

```

## HIF1A
```{r remap_hif1a}
remap_hif1a <- paste0(chipdir,"/hif1a")
n <- "remap_hif1a"
list.files(remap_hif1a)

run <- "ReMap_hif1a"
name <- paste(run,"_",sep="")

beddir <- paste(remap_hif1a)

beds <- list.files(beddir, pattern=".bed.gz")
beds <- file.path(beddir,beds)

peak <- readPeakFile(beds[2])
length(peak)

peak_table <- {}
i <- beds[1]
for (i in beds) {
peak <- readPeakFile(i)
n <- basename(i) %>% str_remove(pattern=".bed.gz")
# print(length(peak))
peak_table <- rbind(peak_table,data.frame(sample = n, peaks = length(peak)))
}
peak_table


promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
genebody <- getBioRegion(TxDb = txdb,
                         by = "gene",
                         type = "body",
                         upstream = rel(1),
                         downstream = rel(1))

file.exists(beds) %>% summary()
peak_list <- lapply(beds,readPeakFile)
peak_list <- lapply(peak_list,keepStandardChromosomes,pruning.mode="coarse")
names <- paste(name,str_split(basename(beds),pattern="\\.", simplify = T)[,3], sep="") %>% str_remove(pattern="_hypoxia|_HYPOX|_HYPO")
names(peak_list) <- names

tagMatrixList_proms <- lapply(peak_list, getTagMatrix, windows=promoter)

# tagMatrixList_genes <- lapply(peak_list, getTagMatrix, windows = genebody, nbin = 100,
#                              upstream = rel(1),downstream = rel(1))

prom1 <- plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000))+ggtitle(run)
prom2 <- plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), facet="row")+ggtitle(run)
prom1+prom2
tagHeatmap(tagMatrixList_proms[3:4])

peakAnnoList <- lapply(peak_list, annotatePeak, TxDb=txdb,
                          tssRegion=c(-3000, 3000), verbose=TRUE)

plotAnnoBar(peakAnnoList)+ggtitle(run)


```

## Annotate peaks
```{r remap_hif1a_annotate}

remap_hif1a_peaks_anno_table <- list()
remap_hif1a_genes_list <- list()
for (i in 1:length(peakAnnoList)){
  print(paste(i,": ",names(peakAnnoList)[i],sep=""))
  n <- names(peakAnnoList[i])
  table <- as.data.frame(peakAnnoList[[i]]@anno)
  table$transcriptId2  <- sub("\\.\\d+$", "", table$transcriptId)
  table$entrez <- table$geneId
  table$geneId <- mapIds(edb, keys = table$transcriptId2, column = "GENEID", keytype = "TXID")
  table$symbol <- mapIds(edb, keys = table$transcriptId2, column = "SYMBOL", keytype = "TXID")
  assign(paste("narrow_table",n,sep="_"),table)
  table_filter <- subset(table, annotation != "Distal Intergenic")
  table_filter$annotation_short <-  str_split(table_filter$annotation,pattern = " ", simplify = TRUE)[,1]

  print(paste("Before: ",length(rownames(table)),", After: ",length(rownames(table_filter))," (",round(length(rownames(table_filter))/length(rownames(table))*100),"%)"))
  remap_hif1a_peaks_anno_table[[n]] <- table_filter
  remap_hif1a_genes_list[[n]] <- table_filter$geneId %>% unique()
}


save(remap_hif1a_peaks_anno_table,file="remap_hif1a_peaks_anno_table.peaks")

# write_xlsx(npeakAnnoList_table[["9929"]], path="9929_HIF1A_peaks.xlsx")


```


# SK-N-BE(2)
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167477
## Load Peaks
```{r sknbe2}
run <- "SKNBE2_HIF1A"
list.dirs(chipdir)
SKNBE2.dir <- paste0(chipdir,"SKNBE2")
list.files(SKNBE2.dir)
SKNBE2.file <- paste0(SKNBE2.dir,"/GSE167477_RAW.tar")
file.exists(SKNBE2.file)
untar(SKNBE2.file,exdir=SKNBE2.dir)
list.files(SKNBE2.dir)

beddir <- paste(SKNBE2.dir)

extraCols_narrowPeak <- c(signalValue = "numeric", pValue = "numeric",
                          qValue = "numeric", peak = "integer")



beds <- list.files(SKNBE2.dir, pattern=".narrowPeak")
beds <- file.path(SKNBE2.dir,beds)
file.exists(beds) %>% summary()

peak <- readPeakFile(beds[4])
length(peak)

peak_table <- {}
npeak_list <- {}
diff_peaks_list <- {}
i <- beds[1]
for (i in beds) {
  print(i)
npeak <- import(i, format = "BED",extraCols = extraCols_narrowPeak)
n <- paste0("SKNBE2_",str_split(basename(i),pattern="_", simplify = T)[,3])
npeak1 <- keepStandardChromosomes(npeak, pruning.mode="coarse")
filtered2 <- findOverlaps(npeak1, bl)
npeak2 <- npeak1[-from(filtered2)] 
npeak3 <- npeak2 %>% unique()
peak_table <- rbind(peak_table,data.frame(sample = n,
                                          peaks = length(npeak),
                                          peaks_Chr = length(npeak1),
                                          peaks_bl = length(npeak2),
                                          peaks_unique = length(npeak3)))
npeak_list[[n]] <- npeak3
}
peak_table %>% kable()

npeak_list

# Get genomic regions
promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
genebody <- getBioRegion(TxDb = txdb,
                         by = "gene",
                         type = "body",
                         upstream = rel(1),
                         downstream = rel(1))



# peak_list <- lapply(beds,readPeakFile)
# peak_list <- lapply(peak_list,keepStandardChromosomes,pruning.mode="coarse")


tagMatrixList_proms <- lapply(npeak_list, getTagMatrix, windows=promoter)

prom1 <- plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000))+ggtitle(run)
prom2 <- plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), facet="row")+ggtitle(run)
prom1+prom2
tagHeatmap(tagMatrixList_proms[3:4])

peakAnnoList <- lapply(npeak_list, annotatePeak, TxDb=txdb,
                          tssRegion=c(-3000, 3000), verbose=TRUE)

plotAnnoBar(peakAnnoList)+ggtitle(run)

```
## Substract Nx Peaks
```{r sknbe2_hx_nx}

# Substract Hx vs. Nx
names(npeak_list)
Hx_samples <- c("SKNBE2_H1", "SKNBE2_H2", "SKNBE2_H3")
Nx_samples <- c("SKNBE2_N1", "SKNBE2_N2", "SKNBE2_N3")

hx_peak_list <- list()
hx_peak_table <- {}
for (i in 1:length(Hx_samples)){
  hx <- Hx_samples[i]
  nx <- Nx_samples[i]
  all_peaks <- npeak_list[[hx]]
  nx_peaks <- npeak_list[[nx]]
  filtered <- findOverlaps(all_peaks, nx_peaks)
  keep <- from(filtered) %>% unique() %>% -.
  hx_peaks <- all_peaks[keep]
  
  peak_numbers <- data.frame(sample = Hx_samples[i],
                                                  Hx_peaks = length(all_peaks),
                                                  Nx_peaks = length(nx_peaks),
                                                  Hx_Nx_peaks = length(hx_peaks))
  # print(peak_numbers)
  hx_peak_table <- rbind(hx_peak_table,peak_numbers)
  hx_peak_list[[i]] <- hx_peaks %>% unique()
}

hx_peak_table %>% kable()

names(hx_peak_list) <- Hx_samples

```

## Venn Peaks
```{r sknbe2_venn_peaks}
input_list <- hx_peak_list
n <- "SKNBE2"
l <- length(input_list)

olvenn <- findOverlapsOfPeaks(input_list)

venn <- makeVennDiagram(olvenn,
                # NameOfPeaks= names(input_list),
                fill=viridis(10)[4:6],
                cat.col=viridis(10)[4:6],
                # cat.pos = c(0,0, -70,+70,0),
                # cat.dist = c(0.2,0.2,0.23,0.22,0.2),
                disable.logging = TRUE,
                filename=NULL,
                title=n
                )

```

## Annotate peaks
```{r sknbe2_annotate}

SKNBE2_peaks_anno <- lapply(hx_peak_list, annotatePeak, TxDb=txdb,
                          tssRegion=c(-3000, 3000), verbose=TRUE)

keytypes(edb)

SKNBE2_peaks_anno_table <- list()
SKNBE2_genes_list <- list()
for (i in 1:length(SKNBE2_peaks_anno)){
  print(paste(i,": ",names(SKNBE2_peaks_anno)[i],sep=""))
  n <- names(SKNBE2_peaks_anno[i])
  table <- as.data.frame(SKNBE2_peaks_anno[[i]]@anno)
  table$transcriptId2  <- sub("\\.\\d+$", "", table$transcriptId)
  table$entrez <- table$geneId
  table$geneId <- mapIds(edb, keys = table$transcriptId2, column = "GENEID", keytype = "TXID")
  table$symbol <- mapIds(edb, keys = table$transcriptId2, column = "SYMBOL", keytype = "TXID")
  assign(paste("narrow_table",n,sep="_"),table)
  table_filter <- subset(table, annotation != "Distal Intergenic")
  table_filter$annotation_short <-  str_split(table_filter$annotation,pattern = " ", simplify = TRUE)[,1]

  print(paste("Before: ",length(rownames(table)),", After: ",length(rownames(table_filter))," (",round(length(rownames(table_filter))/length(rownames(table))*100),"%)"))
  SKNBE2_peaks_anno_table[[n]] <- table_filter
  SKNBE2_genes_list[[n]] <- table_filter$geneId %>% unique()
}

save(SKNBE2_peaks_anno_table,file="SKNBE2_peaks_anno_table.peaks")

# write_xlsx(npeakAnnoList_table[["9929"]], path="9929_HIF1A_peaks.xlsx")

```

## Venn Genes
```{r sknbe2_venn_genes}

input_list <- SKNBE2_peaks_anno_table %>% lapply(.,'[',,"geneId") %>% lapply(.,unique)
n <- "SKNBE2"
l <- length(input_list)

plt1 <- venn.diagram(
    x = input_list,
    fill = viridis(10)[4:6],
    main.fontface = "bold",
    fontfamily ="Arial",
    category.names = paste(names(input_list),"\n(",input_list %>% summary() %>% .[c(1:length(input_list))],")",sep=""),
    force.unique = TRUE, na = "remove", total.population = TRUE,
    filename = NULL,
    lwd = 2,
    lty = 'blank',
    cat.fontface = "bold",
    cat.fontfamily = "arial")

patchwork::wrap_elements(plt1)


```


# Schödel
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE167477
## Load Peaks
```{r schoedel, eval=FALSE}
run <- "SKNBE2_HIF1A"
list.dirs(chipdir)
SKNBE2.dir <- paste0(chipdir,"SKNBE2")
list.files(SKNBE2.dir)
SKNBE2.file <- paste0(SKNBE2.dir,"/GSE167477_RAW.tar")
file.exists(SKNBE2.file)
untar(SKNBE2.file,exdir=SKNBE2.dir)
list.files(SKNBE2.dir)

beddir <- paste(SKNBE2.dir)

extraCols_narrowPeak <- c(signalValue = "numeric", pValue = "numeric",
                          qValue = "numeric", peak = "integer")

beds <- list.files(SKNBE2.dir, pattern=".narrowPeak")
beds <- file.path(SKNBE2.dir,beds)

peak <- readPeakFile(beds[4])
length(peak)

peak_table <- {}
npeak_list <- {}
diff_peaks_list <- {}
i <- beds[1]
for (i in beds) {
  print(i)
npeak <- import(i, format = "BED",extraCols = extraCols_narrowPeak)
n <- paste0("SKNBE2_",str_split(basename(i),pattern="_", simplify = T)[,3])
npeak1 <- keepStandardChromosomes(npeak, pruning.mode="coarse")
filtered2 <- findOverlaps(npeak1, bl)
npeak2 <- npeak1[-from(filtered2)] 
npeak3 <- npeak2 %>% unique()
peak_table <- rbind(peak_table,data.frame(sample = n,
                                          peaks = length(npeak),
                                          peaks_Chr = length(npeak1),
                                          peaks_bl = length(npeak2),
                                          peaks_unique = length(npeak3)))
npeak_list[[n]] <- npeak3
}
peak_table %>% kable()


promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)
genebody <- getBioRegion(TxDb = txdb,
                         by = "gene",
                         type = "body",
                         upstream = rel(1),
                         downstream = rel(1))

file.exists(beds) %>% summary()

peak_list <- lapply(beds,readPeakFile)
peak_list <- lapply(peak_list,keepStandardChromosomes,pruning.mode="coarse")

names <- paste0("SKNBE2_",str_split(basename(beds),pattern="_", simplify = T)[,3])
names(peak_list) <- names

tagMatrixList_proms <- lapply(peak_list, getTagMatrix, windows=promoter)

prom1 <- plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000))+ggtitle(run)
prom2 <- plotAvgProf(tagMatrixList_proms, xlim=c(-3000, 3000), facet="row")+ggtitle(run)
prom1+prom2
tagHeatmap(tagMatrixList_proms[3:4])+ggtitle(run)

peakAnnoList <- lapply(peak_list, annotatePeak, TxDb=txdb,
                          tssRegion=c(-3000, 3000), verbose=TRUE)

plotAnnoBar(peakAnnoList)+ggtitle(run)


```

