---
title: "Figures"
author: "Kelterborn"
date: "2024-07-18"
output:
  github_document:
    html_preview: false
    toc: true
    toc_depth: 2
always_allow_html: true
editor_options: 
  chunk_output_type: inline
knit: (function(input_file, encoding) {
    rmarkdown::render(input_file,output_file= 'Readme.md')
        })
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = T,
                      error=FALSE,
                      warning=FALSE,
                      message=FALSE,
                      dpi=150)

setwd(dirname(rstudioapi::getActiveDocumentContext()$path))

```

# 0. Load

## - Load R librarys

```{r R_update, eval=FALSE, include=FALSE}
BiocManager::install()
BiocManager::valid()
# update.packages(oldPkgs = old.packages())
BiocManager::install("ggExtra")
install.packages("reticulate")


```

```{r librarys, include=FALSE}

library(DESeq2)
library(GEOquery)
library(tidyverse)
library(CorLevelPlot)
library(gridExtra)
library(viridis)

library(clusterProfiler)
library(ggupset)
library(enrichplot)
library(stats)
library(cowplot)
library(VennDiagram)
library(ggplot2)
library(stringr)
library(rlang)
library(reshape2)
library(readxl)
library(SummarizedExperiment)
library(biomaRt)
library(org.Hs.eg.db)
library(Rfast)
library(xlsx)
library(EnhancedVolcano)
library(PCAtools)
library(rgl)
library(impute)
library(magrittr)     
library(flashClust)
library(AnnotationHub)
library(venn)

library(gridExtra)
library(patchwork)
library(grid)
library(reshape2)

library(ComplexHeatmap)
# library(pheatmap)
library(plyr)
library(ggExtra)

library(kableExtra)
library(knitr)
library(RColorBrewer)
library(circlize)
library(devtools)

library(plotly) 
library(stats) 
library(tsne)
library(umap)
library(Rtsne)
library(data.table)

requireNamespace('png', quietly = TRUE)

ifelse(Sys.info()["sysname"]== "Linux",
       s <- "/mnt/s",
       s <- "S:")
dir <- paste(s,"AG/AG-Scholz-NGS/Daten/Simon/RNA-Seq_Kelly_all",sep="/")
gitdir <- paste(dir,"git_RNAseq_Kelly_Hx",sep="/")
data <- paste(dir,"data",sep="/")
dgedir <- paste(gitdir,"2B_DGE",sep="/")

par(mfrow = c(1,1))

futile.logger::flog.threshold(futile.logger::ERROR, name = "VennDiagramLogger")

```

## - Load dds

```{r dds, include=FALSE}

load(file=paste(data,"deseq2_treatment.dds", sep="/"))
dds_t <- dds
load(file=paste(data,"deseq2_condition.dds", sep="/"))
dds_c <- dds
load(file=paste(data,"deseq2_experiment_wgcna.dds", sep="/"))
dds_e <- dds

# load(file=paste(data,"deseq2_experiment_wgcna.dds", sep="/"))
load(file=paste(data,"deseq2_wgcna.dds", sep="/"))

source_file <- dirname(rstudioapi::getSourceEditorContext()$path)
setwd(source_file)

```

## - Colour sheme

```{r colors, echo=FALSE}

colors_paired <- c("#A6CEE3", "#1F78B4", "#B2DF8A", "#33A02C", "#FB9A99", "#E31A1C", "#FDBF6F", "#FF7F00")
colors_v <- c("#440154FF", "#482878FF", "#3E4A89FF", "#31688EFF", "#26828EFF", "#1F9E89FF", "#35B779FF", "#6DCD59FF", "#B4DE2CFF", "#FDE725FF")
colors <- c("lavenderblush3","lavenderblush4","#90caf9","#1976d2", "#82e0aa", "#239b56", "#f8c471", "#b9770e") 
colors4 <- colors[c(1,3,5,7)]
# Volcano colors
colors_vul <- c(colors[4:3],"orangered3","salmon1","hotpink4","hotpink1","darkseagreen4","darkseagreen1")

df <- data.frame(n=c(1:8),
                 color=colors)
```


# Figure 1
## PCA
```{r 1_pca_pre}

vst_dat <- assay(vst(dds))

p <- pca(vst_dat, metadata = colData(dds), removeVar = 0.99)
pca_table <- cbind(p$rotated,p$metadata)
pca1 <- ggplot(pca_table, aes(PC2, PC1, color=genotype, shape=treatment)) +
  geom_hline(yintercept = 0, linewidth = 0.1) + 
  geom_vline(xintercept = 0, linewidth = 0.1) +
  geom_point(size=4, alpha=0.5, stroke=1) +
  labs(title = "top 1% variable genes") +
  ylab(paste0("PC1: ",p$variance["PC1"] %>% round(digits = 1),"% variance")) +
  xlab(paste0("PC2: ",p$variance["PC2"] %>% round(digits = 1),"% variance")) +
  scale_color_manual(values=colors[c(2,4,6,8)]) +
  scale_shape_manual(values = c(21,16)) + 
  scale_fill_manual(values=c(colors[1],"white",colors[2],"white")) +
  theme_bw() +
  scale_y_reverse() +
  removeGrid(x=T, y=T)

p <- pca(vst_dat, metadata = colData(dds), removeVar = 0.95)
pca_table <- cbind(p$rotated,p$metadata)
pca5 <- ggplot(pca_table, aes(PC2, PC1, color=genotype, shape=treatment)) +
  geom_hline(yintercept = 0, linewidth = 0.1) + 
  geom_vline(xintercept = 0, linewidth = 0.1) +
  geom_point(size=4, alpha=0.5, stroke=1) +
  labs(title = "top 5% variable genes") +
  ylab(paste0("PC1: ",p$variance["PC1"] %>% round(digits = 1),"% variance")) +
  xlab(paste0("PC2: ",p$variance["PC2"] %>% round(digits = 1),"% variance")) +
  scale_color_manual(values=colors[c(2,4,6,8)]) +
  scale_shape_manual(values = c(21,16)) + 
  scale_fill_manual(values=c(colors[1],"white",colors[2],"white")) +
  theme_bw() +
  scale_y_reverse() +
  removeGrid(x=T, y=T)

p <- pca(vst_dat, metadata = colData(dds), removeVar = 0.90)
pca_table <- cbind(p$rotated,p$metadata)
pca10 <- ggplot(pca_table, aes(PC2, PC1, color=genotype, shape=treatment)) +
  geom_hline(yintercept = 0, linewidth = 0.1) + 
  geom_vline(xintercept = 0, linewidth = 0.1) +
  geom_point(size=4, alpha=0.5, stroke=1) +
  labs(title = "top 10% variable genes") +
  ylab(paste0("PC1: ",p$variance["PC1"] %>% round(digits = 1),"% variance")) +
  xlab(paste0("PC2: ",p$variance["PC2"] %>% round(digits = 1),"% variance")) +
  scale_color_manual(values=colors[c(2,4,6,8)]) +
  scale_shape_manual(values = c(21,16)) + 
  scale_fill_manual(values=c(colors[1],"white",colors[2],"white")) +
  theme_bw() +
  scale_x_reverse() +
  removeGrid(x=T, y=T)


```
```{r, 1_pca_plot_variants, fig.width=10, fig.height=4}
pca1+pca5+pca10+ plot_layout(guides = "collect", axes="collect", axis_titles="collect") & 
  theme(legend.position = 'bottom')
```

```{r 1_pca_plot_final, fig.height=4, fig.width=5, dpi=300, out.width="50%"}
pca1

```


# Figure 2
## Volanos
```{r 2_volcanos}


```

